lib/traject/macros/marc21_semantics.rb

```diff
@@ -327,7 +327,11 @@ def self.publication_date(record, estimate_tolerance = 15, min_year = 500, max_y
       if field008 && field008.length >= 11
         date_type = field008.slice(6)
         date1_str = field008.slice(7,4)
-        date2_str = field008.slice(11, 4) if field008.length > 15
+        if field008.length > 15
+          date2_str = field008.slice(11, 4)
+        else
+          date2_str = date1_str
+        end
 
         # for date_type q=questionable, we have a range.
         if (date_type == 'q')
```

Comment:
With your changes this should probably  check for `'q'` OR `date1_str == date2_str`.

Reply:
Understood. Please allow me some time, as I don't have practice with tests, and I have to learn how to do them.

Reply:
I've done my best, please bear with me. This has been my (arduous) first test task.  I haven't been able to test the test itself, sorry; it may be something to do with an orphan `/usr/local/bin/rake` that cannot find its ruby interpreter (I'm also quite new dealing different ruby interpreters).  However, I expect the Marc record to be correct.

Anyway, to my understanding the condition it reads better this way:

```
 if date_type == 'q' and date1_str != date2_str
```
and that's why I've put it this way. However, I'm ready to listen to counter arguments, thanks